### DISCUSSION TITLE
Button Messages Cookbook

### DISCUSSION THREAD

**Discussion by smashah** [open-wa/wa-automate-nodejs#2669] [General]
https://docs.openwa.dev/classes/api_Client.Client.html#B

1. Normal text message
```js
await client.B(chatId, {
    "conversation": "test"
});
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173396070-5b51b7cb-1094-4ab1-b466-faf5ec9babbd.png)
</details>

2. Buttons

```js
await client.sendButtons(chatId, "a", [
{
id: "2",
"text": "Click"
},
{
id: "3",
"text": "Call"
},
{
id: "1",
"text": "Open Link"
}],"b","c")
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173396197-85b49254-4e3a-4c1d-9500-3548b00b33b8.png)
</details>

3. List message

```js
await client.B(chatId, {
            "listMessage":  {
                "title": "TITLE",
                "description": "DESC",
                "buttonText": "BUTTON",
                "listType": 1, //UNKNOWN = 0; SINGLE_SELECT = 1; PRODUCT_LIST = 2;
                "sections": [
                    {
                        "rows":[{
                            "title": `TITLE 1`,
                            "description": "DESC 1",
                            "rowId": "ROW 1"
                          },{
                            "title": "TITLE 2",
                            "description": "DESC 2",
                            "rowId": "ROW 2"
                          }
                        ]
                    }
                ]
            }
});
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173396396-f5a28c41-e6b5-4ac1-9837-3ab6191924cd.png)
![image](https://user-images.githubusercontent.com/5903821/173398187-f7ed36f9-577e-485d-8b99-9d6d1f6c351a.png)

</details>

4. Advanced Buttons
```js
await client.B(chatId, {
    "templateMessage": {
        "hydratedTemplate": {
            "hydratedButtons": [
                {
                    "index": 1,
                    "urlButton": {
                        "displayText": "I am a link",
                        "url": "https://github.com/open-wa/wa-automate-nodejs"
                    }
                },
                {
                    "index": 2,
                    "callButton": {
                        "displayText": "Call me now!",
                        "phoneNumber": "+1234567890"
                    }
                },
                {
                    "index": 3,
                    "quickReplyButton": {
                        "displayText": "Or just click me",
                        "id": "button-1"
                    }
                }
            ],
            "hydratedContentText": "content",
            "hydratedFooterText": "footerer"
        }
    }
});
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173398862-87713bae-cef7-42a1-b8f6-459017e2edfb.png)

</details>

5. Location Buttons
```js
await client.sendButtons(chatId, {lat:21.5036885,lng:-21.4358053,caption:"Is this your location?"}, [
{
id: "2",
"text": "Click"
},
{
id: "3",
"text": "Call"
},
{
id: "1",
"text": "Open Link"
}])
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173399085-91eb36bc-2c98-481a-83b1-5a4b7273c980.png)

</details>

6. Image Buttons

```js
await client.sendAdvancedButtons(chatId, "https://cdn-prod.medicalnewstoday.com/content/images/articles/322/322868/golden-retriever-puppy.jpg", [
{
id: "2",
"text": "Click"
},
{
id: "3",
"text": "Call"
},
{
id: "1",
"text": "Open Link"
}], "This is an image button message", "basic buttons", "img.jpeg")
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173408243-af6318ce-62a0-4762-8182-a9083328df32.png)

</details>

7. Image buttons (url/call)

```js
await client.sendAdvancedButtons(chatId, "https://cdn-prod.medicalnewstoday.com/content/images/articles/322/322868/golden-retriever-puppy.jpg", [
{
url: "https://www.google.com",
"text": "Link"
},
{
number: "44771234123423",
"text": "Call"
},
{
id: "1",
"text": "Open Link"
}], "This is an image button message", "with advanced buttons", "img.jpeg")
```

<details>
  <summary>Result</summary>

![image](https://user-images.githubusercontent.com/5903821/173409100-659a9d96-320d-4190-933d-8aa74a0f2b6c.png)

</details>

**Comment by cleancode98** [comment]
Hello! Is advanced buttons working on Multi-Device WhatsApp Business numbers?

**Reply by smashah** [reply]
Yes

**Reply by cleancode98** [reply]
Thanks! Thinking of purchasing the license, can you please clarify if the `wapi.B()` depends on `Baileys` package? Or is it an standalone implementation that relies on whatsapp web native code?

**Reply by smashah** [reply]
@cleancode98 B uses the internal session websocket to attempt to send raw payloads so it functions similarly to Baileys but just VIA the existing session. Baileys is not a dependency.

**Comment by fjbotto** [comment]
Are these Button Messages unlocked when purchasing a license?

**Reply by smashah** [reply]
Yes

**Comment by infiniziconsulting** [comment]
Hello, I was using sendListMessage function till end of May but now when I renewed my subscription, it is not working.

**Comment by infiniziconsulting** [comment]
Is it possible to tell us how to remove licence restriction so that we can use the complete code without any hiccups?
